monkey/agent_plugins/credentials_collectors/chrome/src/linux_credentials_database_selector.py

```diff
@@ -1,11 +1,75 @@
-from typing import Collection
+import logging
+import pwd
+from pathlib import Path
+from typing import Collection, Dict, Sequence, Set
 
 from .utils import BrowserCredentialsDatabasePath
 
+logger = logging.getLogger(__name__)
 
-class LinuxCredentialsDatabaseSelector:
-    def __init__(self):
-        pass
+LinuxUsername = str
+UserDirectories = Dict[LinuxUsername, Path]
+
+# If we don't use a password manager to store an auto-fill password
+# then Linux uses a default master key to "encrypt" the passwords
+# which funny enough it is 'peanuts'
+DEFAULT_MASTER_KEY = "peanuts".encode()
+
+LOGIN_DATABASE_FILENAME = "Login Data"
+CHROMIUM_BASED_DB_PATHS = [
+    ("Google Chrome", ".config/google-chrome"),
+    ("Chromium", ".config/chromium"),
+]
 
+
+class LinuxCredentialsDatabaseSelector:
     def __call__(self) -> Collection[BrowserCredentialsDatabasePath]:
-        return []
+        database_paths: Set[BrowserCredentialsDatabasePath] = set()
+        for browser_database_path in self._get_browser_database_paths():
+            try:
+                login_data_paths = self._get_login_data_paths(browser_database_path)
+                database_paths.update(login_data_paths)
+            except Exception:
+                logger.debug("Failed to get login data paths for: {browser_database_path}")
+
+        return database_paths
+
+    def _get_browser_database_paths(self) -> Sequence[Path]:
+        database_paths = []
+
+        for username, home_dir_path in self._get_home_directories().items():
+            for browser_name, browser_path in CHROMIUM_BASED_DB_PATHS:
+                browser_db_path = home_dir_path / browser_path
+                try:
+                    if browser_db_path.exists():
+                        database_paths.append(browser_db_path)
+                except Exception as err:
+                    logger.exception(
```

Comment:
This logs at the error level. Does the average user really need an error message?